src/BuildScriptGeneratorCli/PipDownloadEventLogger.cs

```diff
@@ -0,0 +1,57 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    internal class PipDownloadEventLogger : ITextStreamProcessor
+    {
+        private const string PipDownloadMarker = "Downloading";
+        private const string EventName = "PipDownload";
+
+        private readonly ILogger _logger;
+
+        private EventStopwatch _currentDownload = null;
+
+        public PipDownloadEventLogger(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void ProcessLine(string line)
+        {
+            if (_currentDownload != null)
+            {
+                _currentDownload.AddProperty("nextLine", line.Trim());
+                _currentDownload.Dispose();
+                _currentDownload = null;
+            }
+
+            if (line.Contains(PipDownloadMarker))
```

Comment:
"Downloading" seems too broad... isn't there a regex that we can use instead?

Reply:
This seems to be enough - especially considering *this checker only runs for Python builds*.
We could come up with a regex, but my guess is that it'll be difficult to write and to maintain